conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/OptionsHandler.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.google.common.base.Joiner;
+import com.palantir.logsafe.Preconditions;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.Headers;
+import io.undertow.util.HttpString;
+import io.undertow.util.Methods;
+import io.undertow.util.StatusCodes;
+import java.util.Set;
+
+final class OptionsHandler implements HttpHandler {
+
+    private final String allowValue;
+
+    OptionsHandler(Set<HttpString> methods) {
+        this.allowValue = Joiner.on(", ").join(methods);
+    }
+
+    @Override
+    public void handleRequest(HttpServerExchange exchange) {
+        Preconditions.checkState(Methods.OPTIONS.equals(exchange.getRequestMethod()), "Expected an OPTIONS request");
+        exchange.getResponseHeaders().put(Headers.ALLOW, allowValue);
+        exchange.setStatusCode(StatusCodes.NO_CONTENT);
```

Comment:
Do we want `Access-Control-Max-Age` too? https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request

Reply:
If we want to set Access-Control headers, I think that belongs in [WebSecurityHandler](https://github.com/palantir/conjure-java/blob/4caf6a593d19597312f7895f41230aa2859589fe/conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/WebSecurityHandler.java). I don't think there's anything specific to options requests for access control headers beyond standard browser use, but I'm not completely confident on that.